js/content/common.js

```diff
@@ -196,16 +196,45 @@ let ExtensionLayer = (function() {
 
     // NOTE: use cautiously!
     // Run script in the context of the current tab
-    self.runInPageContext = function(fun){
-        let script  = document.createElement('script');
-        script.textContent = '(' + fun + ')();';
+    self.runInPageContext = function(fun) {
+        let script  = document.createElement("script");
+        script.textContent = '(' + fun + ")();";
         document.documentElement.appendChild(script);
         script.parentNode.removeChild(script);
     };
 
     return self;
 })();
 
+class Messenger {
+    static sendMessage(msgID, info) {
+        window.postMessage({
+            type: "es_" + msgID,
+            information: info
+        }, '*');
+    }
+
+    static addMessageListener(msgID, fn, autoRemove) {
+        let callback = function(e) {
+            if (e.source !== window) { return; }
+            if (!e.data.type) { return; }
```

Comment:
`e.data.type` has the potential to throw if someone does something silly like: `window.postMessage(null)`. Maybe check `e.data` is meaningful?

Reply:
Better safe than sorry I guess